qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/QuteSupportForTemplate.java

```diff
@@ -477,4 +482,126 @@ public WorkspaceEdit generateMissingJavaMember(GenerateMissingJavaMemberParams p
+			return null;
+		}
+	}
+
+	private String getJavadoc(IType type, DocumentFormat documentFormat, String memberName, String signature, IJDTUtils utils, IProgressMonitor monitor, Set<IType> visited) throws JavaModelException {
+		if (visited.contains(type)) {
+			return null;
+		}
+		visited.add(type);
+		if (monitor.isCanceled()) {
+			throw new OperationCanceledException();
+		}
+
+		ITypeResolver typeResolver = createTypeResolver(type);
+
+		// 1) Check the fields for the member
+
+		// Standard fields
+		IField[] fields = type.getFields();
+		for (IField field : fields) {
+			if (isValidField(field, type)
+					&& memberName.equals(field.getElementName())
+					&& signature.equals(typeResolver.resolveFieldSignature(field))) {
+				String javadoc = utils.getJavadoc(field, documentFormat);
+				if (javadoc != null) {
+					return javadoc;
+				}
+			}
+		}
+
+		// Record fields
+		if (type.isRecord()) {
+			for (IField field : type.getRecordComponents()) {
+				// All record components are valid
+				if (memberName.equals(field.getElementName())
+						&& signature.equals(typeResolver.resolveFieldSignature(field))) {
+					String javadoc = utils.getJavadoc(field, documentFormat);
+					if (javadoc != null) {
+						return javadoc;
+					}
+				}
+			}
+		}
+
+		// 2) Check the methods for the member
+		IMethod[] methods = type.getMethods();
+		for (IMethod method : methods) {
+			if (isValidMethod(method, type)) {
+				try {
+					InvalidMethodReason invalid = getValidMethodForQute(method, type.getFullyQualifiedName());
+					if (invalid == null && (signature.equals(typeResolver.resolveMethodSignature(method)))) {
+						String javadoc = utils.getJavadoc(method, documentFormat);
+						if (javadoc != null) {
+							return javadoc;
+						}
+						// otherwise, maybe a supertype has it
+					}
+				} catch (Exception e) {
+					LOGGER.log(Level.SEVERE,
+							"Error while getting method signature of '" + method.getElementName() + "'.", e);
+				}
+			}
+		}
+
+		// 3) Check the superclasses for the member
+
+		// Collect type extensions
+		List<IType> extendedTypes = null;
+		if (type.isInterface()) {
+			IType[] interfaces = findImplementedInterfaces(type, monitor);
+			if (interfaces != null && interfaces.length > 0) {
+				extendedTypes = Arrays.asList(interfaces);
+			}
+		} else {
+			// ex : String implements CharSequence, ....
+			ITypeHierarchy typeHierarchy = type.newSupertypeHierarchy(monitor);
+			IType[] allSuperTypes = typeHierarchy.getSupertypes(type);
+			extendedTypes = Arrays.asList(allSuperTypes);
+		}
+
+		if (extendedTypes != null) {
+			for (IType extendedType : extendedTypes) {
+				String javadoc = getJavadoc(extendedType, documentFormat, memberName, signature, utils, monitor, visited);
+				if (javadoc != null) {
+					return javadoc;
+				}
+			}
+		}
+
+		return null;
+
+	}
+
+	private IType getTypeFromParams(String projectUri, String typeName, IProgressMonitor monitor) throws JavaModelException {

```

Comment:
I will change the signature with 

`getTypeFromParams(String typeName, String projectUri, IProgressMonitor monitor) `

to be consistent with other method.